docs/source/start/declarative.rst

```diff
@@ -153,6 +162,62 @@ functions, which communicate through a functional exchange:
 The ``!promise`` tag (and the ``!uuid`` tag as well) can be used anywhere where
 a model object is expected.
 
+Creating new references
+^^^^^^^^^^^^^^^^^^^^^^^
+
+It is important to understand when new model objects are created and when only
+references are added. The following example would create a reference in the
+``.allocated_functions`` attribute of the
+:py:class:`~capellambse.model.layers.la.LogicalComponent` which is also the
+logical ``root_component`` (parent) to the logical ``root_function``:
+
+.. code-block:: yaml
+   :emphasize-lines: 2
+
+   - parent: !uuid 0d2edb8f-fa34-4e73-89ec-fb9a63001440
+     extend:
+       allocated_functions:
+         - !uuid f28ec0f8-f3b3-43a0-8af7-79f194b29a2d
+
+This is caused by the type of relationship (non-
+:py:class:`~capellambse.model.common.accessors.DirectProxyAccessor`) between
+the parent and its ``allocated_functions``.
+
+It is also possible to create references to promised objects, but extra caution
+for declaring ``promise_id`` \s for resolving these promises successfully:
+
+.. code-block:: yaml
+   :emphasize-lines: 5,9
+
+   - parent: !uuid f28ec0f8-f3b3-43a0-8af7-79f194b29a2d
+     extend:
+       functions:
+         - name: The promised one
+           promise_id: promised-fnc
+   - parent: !uuid 0d2edb8f-fa34-4e73-89ec-fb9a63001440
+     extend:
+       functions:
+         - !promise promised-fnc
+
+The ``promise_id`` declaration can also happen after referencing it.
+
+Moving objects
+^^^^^^^^^^^^^^
+
+The following example would move a logical function from underneath a
+:py:class:`~capellambse.model.layers.la.LogicalFunctionPkg` (accessible via
+``functions``) into ``functions`` of the logical ``root_function`` (parent):
+
+.. code-block:: yaml
+
+   - parent: !uuid f28ec0f8-f3b3-43a0-8af7-79f194b29a2d
+     extend:
+       functions:
+         - !uuid 8833d2dc-b862-4a50-b26c-6f7e0f17faef
+
+since the ``functions`` attribute has a parent/children relationship (i.e. the
+:py:class:`~capellambse.model.common.accessors.DirectProxyAccessor` is used).
```

Comment:
This part of the sentence should be above the code block with the beginning of it, otherwise it looks like it was meant to be another sentence.